cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -238,197 +562,65 @@ instance FromJSON ProtocolParameters where
-      , protocolUpdateExtraPraosEntropy   = merge protocolUpdateExtraPraosEntropy
-      , protocolUpdateMaxBlockHeaderSize  = merge protocolUpdateMaxBlockHeaderSize
-      , protocolUpdateMaxBlockBodySize    = merge protocolUpdateMaxBlockBodySize
-      , protocolUpdateMaxTxSize           = merge protocolUpdateMaxTxSize
-      , protocolUpdateTxFeeFixed          = merge protocolUpdateTxFeeFixed
-      , protocolUpdateTxFeePerByte        = merge protocolUpdateTxFeePerByte
-      , protocolUpdateMinUTxOValue        = merge protocolUpdateMinUTxOValue
-      , protocolUpdateStakeAddressDeposit = merge protocolUpdateStakeAddressDeposit
-      , protocolUpdateStakePoolDeposit    = merge protocolUpdateStakePoolDeposit
-      , protocolUpdateMinPoolCost         = merge protocolUpdateMinPoolCost
-      , protocolUpdatePoolRetireMaxEpoch  = merge protocolUpdatePoolRetireMaxEpoch
-      , protocolUpdateStakePoolTargetNum  = merge protocolUpdateStakePoolTargetNum
-      , protocolUpdatePoolPledgeInfluence = merge protocolUpdatePoolPledgeInfluence
-      , protocolUpdateMonetaryExpansion   = merge protocolUpdateMonetaryExpansion
-      , protocolUpdateTreasuryCut         = merge protocolUpdateTreasuryCut
-      }
-      where
-        -- prefer the right hand side:
-        merge :: (ProtocolParametersUpdate -> Maybe a) -> Maybe a
-        merge f = f ppu2 `mplus` f ppu1
-
-instance Monoid ProtocolParametersUpdate where
-    mempty =
-      ProtocolParametersUpdate {
-        protocolUpdateProtocolVersion     = Nothing
-      , protocolUpdateDecentralization    = Nothing
-      , protocolUpdateExtraPraosEntropy   = Nothing
-      , protocolUpdateMaxBlockHeaderSize  = Nothing
-      , protocolUpdateMaxBlockBodySize    = Nothing
-      , protocolUpdateMaxTxSize           = Nothing
-      , protocolUpdateTxFeeFixed          = Nothing
-      , protocolUpdateTxFeePerByte        = Nothing
-      , protocolUpdateMinUTxOValue        = Nothing
-      , protocolUpdateStakeAddressDeposit = Nothing
-      , protocolUpdateStakePoolDeposit    = Nothing
-      , protocolUpdateMinPoolCost         = Nothing
-      , protocolUpdatePoolRetireMaxEpoch  = Nothing
-      , protocolUpdateStakePoolTargetNum  = Nothing
-      , protocolUpdatePoolPledgeInfluence = Nothing
-      , protocolUpdateMonetaryExpansion   = Nothing
-      , protocolUpdateTreasuryCut         = Nothing
-      }
-
+                        <*> o .:? "utxoCostPerByte"
+                        <*> o .:? "costModel" .!= mempty
+                        <*> o .:? "executionUnitPrices" .!= mempty
+                        <*> o .:? "maxTxExecUnits"
+                        <*> o .:? "maxBlockExecUnits"
+                        <*> o .:? "maxValueSize"
+           case eParams of
+             Left err -> fail . Text.unpack $ renderProtocolParamsErr err
+             Right pparams -> return pparams
+
+
+instance IsCardanoEra era => ToJSON (ProtocolParameters era) where
```

Comment:
As mentioned elsewhere, as far as I can see, I think we can stick to the simple style where there is no era param and the To/FromJSON do not try to do anything special for different eras. It's clear from the types which params are optional, and so those can be optional in the parser, and are allowed to be not present in the toJSON too. This is the simple untyped approach.

So I'd expect this to look very similar to the old `ToJSON` and `FromJSON` instances, just with the extra fields. All the extra types (`ExecutionUnits`, `CostModel` etc) are great of course.